### PR TITLE
OD-287 [Fix] Sorting will be working as expected without binding to the local format.

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-287

## Description
Changed sorting by date plugin to work with iso date that is hidden in the data attribute.
All changes were made to meet [these](https://github.com/Fliplet/fliplet-widget-organization-dashboard/pull/40#discussion_r542276233) requirements  

## Screenshots/screencasts
https://share.getcloudapp.com/YEuQOX48

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: vendor/dataTables.datetime-moment.js [hunk 1]
```diff
@@ -1,73 +1,51 @@
...
+        return null;
       }
 
-      if ( d === '' || d === null || d === '—' ) {
-        return 'moment-' + format;
+      // If we get match with sent format we check by momentjs that data attribute 'isodate' is a valid date
+      // If so we set type to our format
+      if (d.match(format)) {
+        return moment(
+          $(d)
```
Comment (squallstar) [inline vendor/dataTables.datetime-moment.js]: why is this repeated in the `span data-isodate` regexp?

Comment (YaroslavOvdii) [reply vendor/dataTables.datetime-moment.js]: @squallstar Do you mean why we are using such a format?
If so we using such a format to determine that we got the needed cell.

File: src/components/tables/DataTableCell.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div>
     <span v-if="!cellValue && cellValue !== 0">—</span>
-    <span v-else-if="cellType === 'date'">{{ this.transformDate(cellValue) }}</span>
+    <span v-else-if="cellType === 'date'" :data-isoDate="cellValue">{{ this.transformDate(cellValue) }}</span>
```
Comment (tonytlwu) [inline src/components/tables/DataTableCell.vue]: We should stick with kebab-case for attributes.

```suggestion
    <span v-else-if="cellType === 'date'" :data-iso-date="cellValue">{{ this.transformDate(cellValue) }}</span>
```

File: vendor/dataTables.datetime-moment.js (hunk 1 above)
Comment (tonytlwu) [reply vendor/dataTables.datetime-moment.js]: I don't understand the method we're using to fix this problem.

https://datatables.net/examples/advanced_init/html5-data-attributes.html suggests that by assigning a value for `data-order`, that value gets used for sorting. So it sounds like all we need to do is to add `:data-order="cellValue"` to the `<td>` then we're sorted.

Comment (YaroslavOvdii) [reply vendor/dataTables.datetime-moment.js]: @tonytlwu For this way to work as you highlighted, we will need to remake how we are creating a table cell. Now we are putting content into the td element with vue v-for cycle, without knowing what data will be inside the td.
But it is possible to use your method with some changes. 
I'll remake the solution in a few moments and add a new commit.

Comment (YaroslavOvdii) [reply vendor/dataTables.datetime-moment.js]: @tonytlwu Sorry for the long waiting. I've remade our solution and removed plugins at all. We don't need them anymore. `data-order` attribute works just fine for our needs.
New [screencast](https://share.getcloudapp.com/KoulAzed)

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: All sorting functionality
Verify: Sorting works correctly for all data types

Squash and merge: Updates sorting to use DataTable's custom order functionality

**Pull request merged**